### PR TITLE
adds confirm delete wallet

### PR DESCRIPTION
**Pull request by Thykof**

### CODE REVIEW

File: internal/handler/wallet/api.go [hunk 1]
```diff
@@ -2,15 +2,16 @@ package wallet
...
 	"github.com/massalabs/thyra-plugin-wallet/api/server/restapi/operations"
+	"github.com/massalabs/thyra-plugin-wallet/pkg/delete"
 	"github.com/massalabs/thyra-plugin-wallet/pkg/password"
 	"github.com/massalabs/thyra-plugin-wallet/pkg/privateKey"
 )
 
 // AppendEndpoints appends wallet endpoints to the API
 // Note: the password prompter is mandatory for sign endpoint
-func AppendEndpoints(api *operations.MassaWalletAPI, passwordPrompter password.Asker, privateKeyPrompter privateKey.Asker) {
+func AppendEndpoints(api *operations.MassaWalletAPI, passwordPrompter password.Asker, privateKeyPrompter privateKey.Asker, delePrompter delete.Confirmer) {
```
Comment (thomas-senechal) [inline internal/handler/wallet/api.go:12]:
```suggestion
func AppendEndpoints(api *operations.MassaWalletAPI, passwordPrompter password.Asker, privateKeyPrompter privateKey.Asker, deletePrompter delete.Confirmer) {
```

File: internal/handler/wallet/api.go [hunk 2]
```diff
@@ -2,15 +2,16 @@ package wallet
...
 	api.RestWalletCreateHandler = operations.RestWalletCreateHandlerFunc(HandleCreate)
-	api.RestWalletDeleteHandler = operations.RestWalletDeleteHandlerFunc(HandleDelete)
+	api.RestWalletDeleteHandler = NewDelete(delePrompter)
```
Comment (thomas-senechal) [inline internal/handler/wallet/api.go:14]:
```suggestion
	api.RestWalletDeleteHandler = NewDelete(deletePrompter )
```

**Review by thomas-senechal — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -48,7 +26,10 @@ STANDALONE=1 go run cmd/massa-wallet/thyra-plugin-wallet.go
 
 The `STANDALONE` environment variable is to run the plugin without Thyra.
 
-**Build:**
+Now navigate into <http://localhost:8080>. Note that some feature will not work if
```
Comment (ASAPSegfault) [inline README.md:29]:
```suggestion
Now navigate into <http://localhost:8080>. Note that some features will not work if
```

File: internal/handler/wallet/delete.go [hunk 4]
```diff
@@ -6,11 +6,20 @@ import (
...
+	"github.com/massalabs/thyra-plugin-wallet/pkg/delete"
 	"github.com/massalabs/thyra-plugin-wallet/pkg/wallet"
 )
 
+func NewDelete(deleteConfirm delete.Confirmer) operations.RestWalletDeleteHandler {
+	return &walletDelete{deleteConfirm: deleteConfirm}
+}
+
+type walletDelete struct {
+	deleteConfirm delete.Confirmer
```
Comment (ASAPSegfault) [inline internal/handler/wallet/delete.go:18]: This field is poorly named. It doesnt bring any more insight than its type on what it is.

File: pkg/delete/confirm.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	result := <-PasswordDeleteDialog(walletName, f.guiApp)
+	return result.Password, result.Err
+}
+
+// NewFynePrompter creates a new password prompter with the given Fyne GUI application.
+func NewFynePrompter(f *fyne.App) *FynePrompter {
+	return &FynePrompter{guiApp: f}
+}
+
+// Verifies at compilation time that FynePrompter implements Asker interface.
```
Comment (ASAPSegfault) [inline pkg/delete/confirm.go:28]:
```suggestion
// Verifies at compilation time that FynePrompter implements Confirmer interface.
```

File: pkg/delete/confirm.go [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+var _ Confirmer = &FynePrompter{}
+
+// PasswordDeleteDialog displays a password dialog with the given nickname and ask confirmation for delete the account.
```
Comment (ASAPSegfault) [inline pkg/delete/confirm.go:31]:
```suggestion
// PasswordDeleteDialog displays a password dialog with the given nickname and ask confirmation to delete the account.
```

**Review by ASAPSegfault — CHANGES_REQUESTED** [review]

**Review by ASAPSegfault — APPROVED** [review]

**Pull request merged**